osu.Game/Screens/OnlinePlay/Lounge/Components/DrawableRoom.cs

```diff
@@ -243,36 +249,71 @@ protected override void LoadComplete()
+
             wrapper.DelayedLoadComplete += _ =>
             {
-                wrapper.FadeInFromZero(200);
+                Debug.Assert(specialCategoryPill != null);
+                Debug.Assert(endDateInfo != null);
+                Debug.Assert(passwordIcon != null);
 
-                roomCategory.BindTo(Room.Category);
-                roomCategory.BindValueChanged(c =>
-                {
-                    if (c.NewValue > RoomCategory.Normal)
-                        specialCategoryPill.Show();
-                    else
-                        specialCategoryPill.Hide();
-                }, true);
-
-                roomType.BindTo(Room.Type);
-                roomType.BindValueChanged(t =>
-                {
-                    endDateInfo.Alpha = t.NewValue == MatchType.Playlists ? 1 : 0;
-                }, true);
+                wrapper.FadeInFromZero(200);
 
-                hasPassword.BindTo(Room.HasPassword);
-                hasPassword.BindValueChanged(v => passwordIcon.Alpha = v.NewValue ? 1 : 0, true);
+                updateRoomName();
+                updateRoomCategory();
+                updateRoomType();
+                updateRoomHasPassword();
             };
+
+            SelectedItem.BindValueChanged(item => background.Beatmap.Value = item.NewValue?.Beatmap, true);
         }
 
-        protected override IReadOnlyDependencyContainer CreateChildDependencies(IReadOnlyDependencyContainer parent)
+        private void onRoomPropertyChanged(object? sender, PropertyChangedEventArgs e)
         {
-            return new CachedModelDependencyContainer<Room>(base.CreateChildDependencies(parent))
+            switch (e.PropertyName)
             {
-                Model = { Value = Room }
-            };
+                case nameof(Room.Name):
+                    updateRoomName();
+                    break;
+
+                case nameof(Room.Category):
+                    updateRoomCategory();
+                    break;
+
+                case nameof(Room.Type):
+                    updateRoomType();
+                    break;
+
+                case nameof(Room.HasPassword):
+                    updateRoomHasPassword();
+                    break;
+            }
+        }
+
+        private void updateRoomName()
+        {
+            if (roomName != null)
+                roomName.Text = Room.Name;
+        }
+
+        private void updateRoomCategory()
+        {
+            if (Room.Category > RoomCategory.Normal)
+                specialCategoryPill?.Show();
+            else
+                specialCategoryPill?.Hide();
+        }
+
+        private void updateRoomType()
+        {
+            if (endDateInfo != null)
+                endDateInfo.Alpha = Room.Type == MatchType.Playlists ? 1 : 0;
+        }
+
+        private void updateRoomHasPassword()
+        {
+            if (passwordIcon != null)
+                passwordIcon.Alpha = Room.HasPassword ? 1 : 0;
         }
```

Comment:
Related to [comment above](https://github.com/ppy/osu/pull/30634#discussion_r1843422762) - I probably would have merged these into one method, from experience. I'm talking about every case where these groups of small `updateX()` methods have materialised; in my experience the efficiency tradeoff has rarely been worth the mental overhead.

Reply:
I expected these PRs to be reviewed part by part and not in the final thing. I won't do the NRT changes next time and keep it to one PR, my bad.

In this case, I wrote a blurb explaining in https://github.com/ppy/osu/pull/30631. Or at least that I'm aware of the separation.

Reply:
> I expected these PRs to be reviewed part by part and not in the final thing

I would have, but given that seemingly a full review of this diff was done already here, it felt like the wrong thing to review piece-by-piece now...?